components/EvergageABTest.tsx

```diff
@@ -40,12 +44,12 @@ export default class EvergageABTest extends React.Component<IEvergageABTestProps
         }
         const { checkForExperience, handleEvent, props : { campaign,  eventPrefix, timeout } } = this;
         window.addEventListener(`${eventPrefix}-${campaign}`, handleEvent);
-        window.onload = function() {
+        window.addEventListener('load', function() {
             window.setTimeout(checkForExperience, timeout);
-        };
+        });
     }
     checkForExperience(){
-        if(this.state.selectedExperience == null){            
+        if(!this.state.experienceReceived && !this.props.supressFallback){            
```

Comment:
experienced Received will be leveraged so that if an experienced is received for a campaign but it is not found in a given instance of a component it will continue, allowing that experience to be mounted elsewhere

Reply:
i would maybe called this userHasReceivedExperience